pkg/eventhandler/eventhandlers_test.go

```diff
@@ -0,0 +1,701 @@
+			[]string{
+				"sh.keptn.event.action.started", "sh.keptn.event.action.finished",
+			},
+		),
+	)
+}
+
+func TestErrorConnectingToK8s(t *testing.T) {
+
+	tests := []struct {
+		name               string
+		silent             bool
+		expectedEventTypes []string
+	}{
+		{
+			name:   "Error Connecting with silent=false",
+			silent: false,
+			expectedEventTypes: []string{
+				"sh.keptn.event.action.started",
+				"sh.keptn.event.action.finished",
+			},
+		},
+		{
+			name:               "Error Connecting with silent=true",
+			silent:             true,
+			expectedEventTypes: []string{},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(
+			test.name, func(t *testing.T) {
+				mockCtrl := gomock.NewController(t)
+				defer mockCtrl.Finish()
+
+				mockFilter := eventhandlerfake.NewMockImageFilter(mockCtrl)
+				mockJobConfigReader := eventhandlerfake.NewMockJobConfigReader(mockCtrl)
+
+				mockJobConfigReader.EXPECT().GetJobConfig().Return(
+					&config.Config{
+						APIVersion: &apiVersion,
+						Actions: []config.Action{
+							{
+								Name: "catch-all action",
+								Events: []config.Event{
+									{
+										Name:     "sh.keptn.event.*",
+										JSONPath: config.JSONPath{},
+									},
+								},
+								Tasks: []config.Task{
+									{
+										Name:  "sample-task",
+										Image: "sampleimage",
+										Cmd:   []string{"echo"},
+										Args:  []string{"Hello", "World"},
+									},
+								},
+								Silent: test.silent,
+							},
+						},
+					},
+					nil,
+				).Times(1)
+
+				mockK8s := eventhandlerfake.NewMockK8s(mockCtrl)
+				connectionError := errors.New("error connecting to k8s cluster")
+				mockK8s.EXPECT().ConnectToCluster().Return(connectionError).Times(1)
+
+				myKeptn, _, mockEventSender, err := initializeTestObjects("../../test-events/action.triggered.json")
+				require.NoError(t, err)
+
+				sut := EventHandler{
+					Keptn:           myKeptn,
+					JobConfigReader: mockJobConfigReader,
+					ServiceName:     "test-jes",
+					JobSettings:     k8sutils.JobSettings{AlwaysSendFinishedEvent: true},
+					ImageFilter:     mockFilter,
+					Mapper:          new(KeptnCloudEventMapper),
+					K8s:             mockK8s,
+				}
+
+				err = sut.HandleEvent()
+
+				// TODO: weird that if connection to k8s fails we don't send back an error from the handling function
+				assert.NoError(t, err)
+				assert.NoError(
+					t, mockEventSender.AssertSentEventTypes(
+						test.expectedEventTypes,
+					),
+				)
+			},
+		)
+	}
+}
+
+type nameMatcher struct {
+	name string
+}
+
+func (am *nameMatcher) Matches(x interface{}) bool {
+	reflectedValue := reflect.ValueOf(x)
+	if reflectedValue.Kind() == reflect.Ptr || reflectedValue.Kind() == reflect.Interface {
+		// if it's a pointer or an interface dereference to get to the struct
+		reflectedValue = reflectedValue.Elem()
+	}
+
+	if am.name != reflectedValue.FieldByName("Name").String() {
+		return false
+	}
+
+	return true
```

Comment:
I think we can simplify this to `return am.name == reflectedValue.FieldByName("Name").String()`